packages/core/src/submodules/client/setFeature.ts

```diff
@@ -24,3 +29,20 @@ export function setFeature<F extends keyof AwsSdkFeatures>(
   }
   context.__aws_sdk_context.features![feature] = value;
 }
+
+/**
+ * @internal
+ *
+ * @returns the credentials with source feature attribution.
+ */
+export function setCredentialFeature<F extends keyof AwsSdkCredentialsFeatures>(
```

Comment:
nit: move to a new file `setCredentialFeature`?

Reply:
done